share/dkg/dkg_test.go

```diff
@@ -68,9 +70,11 @@ func SetupNodes(nodes []*TestNode, c *DkgConfig) {
 }
 
 func SetupReshareNodes(nodes []*TestNode, c *DkgConfig, coeffs []kyber.Point) {
+	nonce, _ := GetNonce()
```

Comment:
handle errors? i would have `GetNonce()` not return a error, but panic if the rand.read errors.

Reply:
I hesitated to do that but foreseeing the integration within drand I wouldn't have to do `recover()` : in drand the beacon chain generation should continue even if the a new resharing fails to proceed, I dont want the daemon to panic because it couldnt gather randomness. Paradoxically Beacon generation doesnt require randomness :)

Reply:
not having a working source of randomness is a failure case that something is "really wrong" with the machine, and i think worthy of a panic

Reply:
It sometimes happens on VMs that the internal entropy source goes down if it's not refreshed enough - drand only cares about that during a resharing - but any problem with source of entropy should in no way trigger a panic. That's just creating a chain of reaction which shouldn't happen: drand can and should continue to function normally, maybe with reduced functionality (no resharing possible) even with a degraded source of entropy. I want to avoid the scenario where so many VMs on a machine request entropy, there's not enough entropy

I know this situation is really wrong but that's not up to drand to take action upon it, since it doesn't depend on it.

SO: what I propose is that I make this panicking, but I insert a defer recover in drand. Would that be okay to you ?

Reply:
yeah. At the kyber level, i want a cryptographic secure nonce. i am not aware of any VMs in practice where in normal operations the crypto rand operation will fail for any reason that is not full-panic-worthy. The golang crypto rand will instead block until more randomness is available rather than failing iirc